tests/nstack.rs

```diff
@@ -0,0 +1,167 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use canonical::{Canon, CanonError};
+use canonical_derive::Canon;
+
+use nstack::NStack;
+
+use core::borrow::Borrow;
+use microkelvin::*;
+
+#[test]
+fn trivial() {
+    let mut nt = NStack::<u32, Cardinality>::new();
+    assert_eq!(nt.pop().unwrap(), None);
```

Comment:
Avoid to use `unwrap`s whenever is possible, even in tests.
Always prefers to returns a result (tests can have `Result` as return types) and use `?`, like in this case.
In the worst case scenario use `expect`s, but it doesn't apply here.

This is valid for the whole test case, not just this line.